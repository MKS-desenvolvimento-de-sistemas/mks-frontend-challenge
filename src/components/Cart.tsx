import { count } from "console";
import React from "react";
import { connect } from "react-redux";
import Product from "../interfaces";
import ProductCardCart from "./ProductCardCart";

type Props = {
  products: Product[];
  cart: ProductCart[];
  value: number;
};

type ProductCart = { productId: string; count: number };

type CartType = { products: ProductCart; value: number };

class Cart extends React.Component {
  render() {
    const { products, cart, value } = this.props as Props;
    return (
      <aside>
        <div className="cart-header">
          <p>Carrinho de Compras</p>
          <button type="button">X</button>
        </div>
        <div className="products">
          {cart.map((item) => {
            const img = products.find(
              (product) => product.id === Number(item.productId)
            )?.photo;
            const title = products.find(
              (product) => product.id === Number(item.productId)
            )?.name;
            const price = products.find(
              (product) => product.id === Number(item.productId)
            )?.price;
            return (
              <ProductCardCart
                key={`${item.productId}-${item.count}`}
                id={item.productId}
                img={img ? img : ""}
                title={title ? title : ""}
                price={price ? price : "0"}
                count={item.count}
              />
            );
          })}
        </div>
        <div className="footer">
          <div className="total">
          <p>Total:</p>
          <p>R${`${value}`}</p>
          </div>
          <button>Finalizar Compra</button>
        </div>
      </aside>
    );
  }
}

const mapStateToProps = (store: {
  products: { products: Product[] };
  cart: CartType;
}) => ({
  products: store.products.products,
  cart: store.cart.products,
  value: store.cart.value,
});

export default connect(mapStateToProps)(Cart);
