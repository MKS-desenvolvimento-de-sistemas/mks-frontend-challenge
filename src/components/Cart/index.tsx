import React from 'react';
import ItemCart from '../ItemCart';
import {
	CartStyleContainer,
	Container,
	CartHeader,
	CartList,
	CartTotalPrice,
	CartFinality,
} from './style';

const Cart = () => (
	<div>
		<CartStyleContainer>
			<Container>
				<CartHeader>
					<h1>Carrinho de compras</h1>
					<button>&#x02A2F;</button>
				</CartHeader>
				<CartList className='paddingLast'>
					<ItemCart />
					<ItemCart />
					<ItemCart />
					<ItemCart />
					<ItemCart />
				</CartList>
			</Container>
		</CartStyleContainer>
		<CartTotalPrice>
			<h1>Total:</h1>
			<h1>R$798</h1>
		</CartTotalPrice>
		<CartFinality>
			<h1>Finalizar Compra</h1>
		</CartFinality>
	</div>
);

export default Cart;
