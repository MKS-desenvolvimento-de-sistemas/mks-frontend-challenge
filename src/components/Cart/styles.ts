import styled from "styled-components";

export const Cart = styled.aside`
  width: 330px;
  height: 100%;
  max-height: 100%;

  position: fixed;
  top: 0;
  right: 0;
  z-index: var(--z-cart);

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;

  background-color: var(--terciary-color);
  box-shadow: -5px 0px 6px rgba(0, 0, 0, 0.13);

  @media (min-width: 1240px) {
    width: 486px;
  }
`;

export const CartHeader = styled.div`
  width: 100%;
  margin: 25px 0 18px 0;
  padding: 0 15px 0 32px;

  display: flex;
  justify-content: space-between;
  align-items: flex-start;

  font-size: 46px;
  color: var(--secondary-color-300);

  h2 {
    font-weight: 700;
    font-size: 27px;
    color: var(--primary-color-100);
  }

  @media (min-width: 1240px) {
    margin: 36px 0 58px 0;
    padding: 0 22px 0 47px;
    font-size: 38px;

    h2 {
      font-size: 27px;
      max-width: 180px;
    }
  }
`;

export const CartList = styled.ul`
  width: 100%;
  padding: 10px 46px 10px 34px;
  flex: 2;
  overflow-y: scroll;

  scrollbar-width: thin;
  scrollbar-color: var(--primary-color-100) var(--terciary-color);

  &::-webkit-scrollbar {
    width: 12px;
  }

  &::-webkit-scrollbar-track {
    background: var(--terciary-color);
  }

  &::-webkit-scrollbar-thumb {
    background-color: var(--primary-color-100);
    border-radius: 20px;
    border: 3px solid var(--terciary-color);
  }

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  gap: 12.95px;

  @media (min-width: 1240px) {
    gap: 28px;
  }
`;

export const CartTotal = styled.div`
  width: 100%;
  padding: 35px 15px 35px 32px;

  display: flex;
  justify-content: space-between;
  align-items: center;

  font-weight: 700;
  font-size: 28px;
  color: var(--primary-color-100);

  @media (min-width: 1240px) {
    padding: 35px 22px 35px 47px;
  }
`;

export const CartButton = styled.button`
  width: 100%;
  padding: 26px 0;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: var(--secondary-color-300);

  font-weight: 700;
  font-size: 20px;
  color: var(--primary-color-100);

  &:hover {
    filter: opacity(0.7);
  }
`;
