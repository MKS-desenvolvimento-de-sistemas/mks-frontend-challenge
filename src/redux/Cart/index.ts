import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { IProductCart, IProduct } from "../../types/api";

const initialState: { products: IProductCart[] | [] } = {
  products: [],
};

const cart = createSlice({
  name: "products",
  initialState: initialState,
  reducers: {
    add(state, action: PayloadAction<IProduct>) {
      const productAlreadyExists = state.products.find(
        (product) => product.id === action.payload.id
      );

      console.log(action.payload);

      if (productAlreadyExists) {
        productAlreadyExists.quantity++;
      } else {
        state.products = [
          ...state.products,
          { ...action.payload, quantity: 1 },
        ];
      }
    },
    removeOne(state, action: PayloadAction<IProduct["id"]>) {
      let filteredProducts: IProductCart[] | [] = [];
      for (let i = 0; i < state.products.length; i++) {
        let product: IProductCart = state.products[i];

        if (product.id === action.payload) {
          if (product.quantity > 1) {
            product.quantity--;
            filteredProducts = [...filteredProducts, product];
          } else {
            product.quantity = 0;
          }
        } else {
          filteredProducts = [...filteredProducts, product];
        }
      }

      state.products = filteredProducts;
    },
    removeAll(state, action: PayloadAction<IProduct["id"]>) {
      const filteredProducts = state.products.filter(
        (product) => product.id !== action.payload
      );

      state.products = filteredProducts;
    },
  },
});

export const { add, removeOne, removeAll } = cart.actions;
export default cart.reducer;
