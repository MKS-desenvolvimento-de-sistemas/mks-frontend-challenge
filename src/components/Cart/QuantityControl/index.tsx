import * as Styles from "./styles";

interface IQuantityControlProps {
  quantity: number;
  increment: () => void;
  decrement: () => void;
}

const QuantityControl = ({
  quantity,
  increment,
  decrement,
}: IQuantityControlProps) => {
  return (
    <Styles.Container>
      <small>Qtd:</small>
      <Styles.Button
        data-testid="decrement-button"
        type="button"
        title="Diminuir Quantidade"
        blocked={quantity <= 1}
        onClick={() => {
          if (quantity > 1) {
            decrement();
          }
        }}
      >
        -
      </Styles.Button>
      <span data-testid="quantity-display">{quantity}</span>
      <Styles.Button
        data-testid="increment-button"
        type="button"
        title="Aumentar Quantidade"
        onClick={increment}
      >
        +
      </Styles.Button>
    </Styles.Container>
  );
};

export default QuantityControl;
