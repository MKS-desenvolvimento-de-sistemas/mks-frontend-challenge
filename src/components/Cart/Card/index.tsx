import * as Styles from "./styles";
import { useState } from "react";

// REDUX IMPORT'S
import { useDispatch } from "react-redux";
import { add, removeOne, removeAll } from "../../../redux/Cart";

import { IProductCart } from "../../../types/api";
import { AiOutlineClose } from "react-icons/ai";
import QuantityControl from "../QuantityControl";
import { Price } from "../../";

const CartCard = (props: IProductCart) => {
  const { id, name, price, photo, quantity } = props;
  const dispatch = useDispatch();

  const increment = () => {
    dispatch(add(props));
  };

  const decrement = () => {
    dispatch(removeOne(id));
  };

  const remove = () => {
    dispatch(removeAll(id));
  };

  return (
    <Styles.Card>
      <Styles.CardRemove
        type="button"
        title="Remover do Carrinho"
        onClick={remove}
      >
        <AiOutlineClose />
      </Styles.CardRemove>
      <Styles.CardImage imageUrl={photo} />
      <Styles.CardInfo>
        <h4>{name}</h4>
        <QuantityControl
          quantity={quantity}
          increment={increment}
          decrement={decrement}
        />
        <Price price={Number(price)} type="cart" />
      </Styles.CardInfo>
    </Styles.Card>
  );
};

export default CartCard;
