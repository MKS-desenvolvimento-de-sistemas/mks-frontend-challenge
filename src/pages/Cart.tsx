import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import styled from "styled-components";
import CartItem from "../CartItem";
import { cartActions } from "../store";

export interface ICartItem {
  id: number;
  name: string;
  description: string;
  photo: string;
  brand: string;
  price: string;
  createdAt: string;
  updatedAt: string;
  quantity: number;
}

const CartStyles = styled.div`
  position: fixed;
  right: 0;
  width: 35%;
  top: 0;
  background-color: #0a5bb8;
  box-shadow: 0 0 9px #4242427b;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  transition: transform 200ms ease-in-out;
  & .title {
    position: absolute;
    top: 2rem;
    left: 2rem;
    color: white;
    width: 7rem;
    font-size: larger;
    font-weight: 700;
  }

  & .sticky-wrapper {
    position: sticky;
    display: flex;
    justify-content: center;
    width: 80%;
  }
  & .selected-items {
    display: flex;
    flex-direction: column;
    gap: 0.7rem;
    position: absolute;
    top: 6.5rem;
    width: 70%;
    max-height: 60%;
    overflow-y: auto;
    overflow-x: hidden;
    padding-right: 1.5rem;
    padding-left: 3rem;
    padding-top: 1rem;
  }
  & .list-header {
    position: sticky;
    top: 0;
    background-color: white;
  }
  & th {
    width: 1rem;
  }
  & .total {
    position: absolute;
    bottom: 5rem;
    font-size: 2rem;
    color: white;
    display: flex;
    justify-content: center;
    width: 100%;
  }

  & .exit-cart {
    position: absolute;
    right: 1rem;
    top: 1rem;
    background-color: black;
    color: white;
    border-radius: 50rem;
    width: 0.5rem;
    display: flex;
    justify-content: center;
    font-weight: 700;
  }
  & .finalizar-compra {
    position: absolute;
    bottom: 0;
    left: 0px;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 4rem;
    background-color: black;
    color: white;
    font-size: 2rem;
    cursor: pointer;
  }
`;

type Props = {
  setShowCart: React.Dispatch<React.SetStateAction<boolean>>;
  showCart: boolean;
};

const Cart = ({ showCart, setShowCart }: Props) => {
  const [cart, setCart] = useState<ICartItem[]>([]);
  const [total, setTotal] = useState(0);
  const items = useSelector((state) => state.items);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!items || items.length === 0) return;
    console.log({ items });

    setCart(addItemsGroupedToCart(items));
  }, [items]);

  useEffect(() => {
    if (!cart || cart.length === 0) return;
    calcTotal();
  }, [cart]);

  const calcTotal = () => {
    const totalArr = [] as number[];
    cart.forEach((item) => {
      totalArr.push(+item.price * item.quantity);
    });

    const sum = totalArr.reduce((acc, cur) => acc + cur, 0) as number;

    setTotal(sum);
  };

  const addItemsGroupedToCart = (values) => {
    // const grouped = values.reduce((r, v, i, a) => {
    //   if (v === a[i - 1]) {
    //     r[r.length - 1].push(v);
    //   } else {
    //     r.push([v]);
    //   }
    //   return r;
    // }, []);

    const grouped = values.reduce((acc, cur, index, arr) => {
      acc[cur.id] = acc[cur.id] || [];
      acc[cur.id].push(cur);

      return acc;
    }, {});

    const finalArr = [] as ICartItem[];

    Object.values(grouped).forEach((group) => {
      const newItem = {
        id: group[0].id,
        name: group[0].name,
        description: group[0].description,
        photo: group[0].photo,
        brand: group[0].brand,
        price: group[0].price,
        createdAt: group[0].createdAt,
        updatedAt: group[0].updatedAt,
        quantity: group.length,
      };
      finalArr.push(newItem);
    });

    return finalArr;

    // return finalArr;
  };

  return (
    <CartStyles
      style={{ transform: showCart ? "translateX(0%)" : "translateX(100%)" }}
    >
      <button onClick={() => setShowCart(false)} className="exit-cart">
        X
      </button>
      <label className="title">
        Carrinho <br />
        de compras
      </label>
      <div className="selected-items">
        {items.length !== 0 && cart.map((item) => <CartItem item={item} />)}
      </div>

      <div className="total">Total: R${items.length > 0 ? total : "0"}</div>
      <div
        onClick={() => {
          setShowCart(false);
          dispatch(cartActions.deleteAllItems());
        }}
        className="finalizar-compra"
      >
        Finalizar Compra
      </div>
    </CartStyles>
  );
};

export default Cart;
