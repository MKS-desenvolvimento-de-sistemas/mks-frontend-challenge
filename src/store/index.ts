import { createSlice } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import { ICartItem } from "../pages/Cart";
const initialState = { items: [] } as any;

const cartSlice = createSlice({
  name: "carrinho",
  initialState,
  reducers: {
    addToCart(state, action) {
      state.items.push(action.payload);
    },

    addAdditional(state, action) {
      if (!state.items.some((item) => item.id === action.payload.id)) return;
      const item = state.items.find((item) => item.id === action.payload.id);

      state.items.push(item);
    },
    decreaseItem(state, action) {
      const remainingItems = state.items.filter(
        (item) => item.id === action.payload.id
      );

      if (remainingItems.length === 1) return;

      const itemIndex = state.items.findIndex(
        (el) => el.id === action.payload.id
      );

      state.items.splice(itemIndex, 1);
    },
    deleteItem(state, action) {
      const arr = state.items.filter((item) => item.id !== action.payload.id);
      state.items = arr;
      // do {
      //   const itemIndex = state.items.findIndex(
      //     (el) => el.id === action.payload.id
      //   );

      //   state.items.splice(itemIndex, 1);
      // } while (state.items.some((item) => item.id === action.payload.id));
    },
    deleteAllItems(state) {
      state.items = [];
    },
  },
});

const store = configureStore({
  reducer: cartSlice.reducer,
});

export const cartActions = cartSlice.actions;

export default store;
