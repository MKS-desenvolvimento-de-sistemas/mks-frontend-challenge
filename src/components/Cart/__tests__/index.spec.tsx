import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import Cart from "../";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../redux/";
import { IProductCart } from "../../../types/api";

const productCartExampleI: IProductCart = {
  id: 1,
  name: "EXEMPLO 1",
  brand: "Exemplo 1",
  description: "Exemplo 1",
  price: "200",
  photo: "/vercel.svg",
  updatedAt: "2022",
  createdAt: "2022",
  quantity: 2,
};

const productCartExampleII: IProductCart = {
  id: 2,
  name: "EXEMPLO 2",
  brand: "Exemplo 2",
  description: "Exemplo 2",
  price: "350",
  photo: "/vercel.svg",
  updatedAt: "2022",
  createdAt: "2022",
  quantity: 1,
};

const cartExample = [productCartExampleI, productCartExampleII];

describe("Cart Component", () => {
  it("should render the cart component correctly", () => {
    const closeCart = jest.fn();
    render(
      <Provider store={store}>
        <Cart cart={cartExample} closeCart={closeCart} />
      </Provider>
    );

    const cart = screen.getByTestId("cart");
    const cartList = screen.getByTestId("cart-list");

    expect(cart).toHaveTextContent("Carrinho de Compras");
    expect(cartList).toHaveTextContent("EXEMPLO 1");
    expect(cartList).toHaveTextContent("EXEMPLO 2");
  });

  it("should render the total price correctly", () => {
    const closeCart = jest.fn();
    render(
      <Provider store={store}>
        <Cart cart={cartExample} closeCart={closeCart} />
      </Provider>
    );

    const cartTotal = screen.getByTestId("cart-total");

    // 200 * 2 (Produto 1) + 350 * 1 (Produto 2)
    expect(cartTotal).toHaveTextContent("R$ 750");
  });

  it("should match the snapshot", () => {
    const closeCart = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Cart cart={cartExample} closeCart={closeCart} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
