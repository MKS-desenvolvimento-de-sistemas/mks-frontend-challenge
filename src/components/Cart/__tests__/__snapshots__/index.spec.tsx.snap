// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component should match the snapshot 1`] = `
<aside
  className="sc-bczRLJ uGUip"
  data-testid="cart"
>
  <div
    className="sc-gsnTZi esggUP"
  >
    <h2>
      Carrinho de Compras
    </h2>
    <svg
      data-testid="cart-close"
      fill="currentColor"
      height="1em"
      onClick={[MockFunction]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
          "cursor": "pointer",
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Fechar Carrinho
      </title>
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
      />
    </svg>
  </div>
  <ul
    className="sc-dkzDqf dAcNYh"
    data-testid="cart-list"
  >
    <li
      className="sc-papXJ gNQOvW"
      data-testid="cart-card"
    >
      <button
        className="sc-jqUVSM iukfUU"
        data-testid="cart-card--remove"
        onClick={[Function]}
        title="Remover do Carrinho"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
          />
        </svg>
      </button>
      <div
        className="sc-iBkjds sc-kDDrLX idbGkR gDlOlb"
        data-testid="cart-card--image"
      />
      <div
        className="sc-iqcoie gCUGTW"
      >
        <h4>
          EXEMPLO 1
        </h4>
        <div
          className="sc-crXcEl hcUnyF"
        >
          <small>
            Qtd:
          </small>
          <button
            className="sc-evZas kedcOT"
            data-testid="decrement-button"
            onClick={[Function]}
            title="Diminuir Quantidade"
            type="button"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            2
          </span>
          <button
            className="sc-evZas kedcOT"
            data-testid="increment-button"
            onClick={[Function]}
            title="Aumentar Quantidade"
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="sc-fEOsli bEDLcJ"
          data-testid="price"
          type="cart"
        >
          R$200
        </div>
      </div>
    </li>
    <li
      className="sc-papXJ gNQOvW"
      data-testid="cart-card"
    >
      <button
        className="sc-jqUVSM iukfUU"
        data-testid="cart-card--remove"
        onClick={[Function]}
        title="Remover do Carrinho"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
          />
        </svg>
      </button>
      <div
        className="sc-iBkjds sc-kDDrLX idbGkR gDlOlb"
        data-testid="cart-card--image"
      />
      <div
        className="sc-iqcoie gCUGTW"
      >
        <h4>
          EXEMPLO 2
        </h4>
        <div
          className="sc-crXcEl hcUnyF"
        >
          <small>
            Qtd:
          </small>
          <button
            className="sc-evZas eETeAU"
            data-testid="decrement-button"
            onClick={[Function]}
            title="Diminuir Quantidade"
            type="button"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            1
          </span>
          <button
            className="sc-evZas kedcOT"
            data-testid="increment-button"
            onClick={[Function]}
            title="Aumentar Quantidade"
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="sc-fEOsli bEDLcJ"
          data-testid="price"
          type="cart"
        >
          R$350
        </div>
      </div>
    </li>
  </ul>
  <div
    className="sc-hKMtZM jFFcXQ"
  >
    <span>
      Total
    </span>
    <span
      data-testid="cart-total"
    >
      R$ 750
    </span>
  </div>
  <button
    className="sc-eCYdqJ iwIllb"
    title="Finalizar Compra"
    type="button"
  >
    Finalizar Compra
  </button>
</aside>
`;
